### DISCUSSION TITLE
How to add user-facing front-end

### DISCUSSION THREAD

**Discussion by Moai-** [keystonejs/keystone#7476] [Questions] [answered]
Hi! Please let me know if this question is in a bad spot. I've been looking everywhere and I can't find the answer.  

How do I add a user-facing front-end to KeystoneJS? I want to use it to make a personal blog. I understand it is headless, but I can't find any information about how to connect it to any sort of head. Is there an example project that includes a user-facing front-end (like a blog site where users can see posts, search them, etc)? This seems really obvious but I cannot for the life of me find the answer anywhere. Please help if you can.

**Comment by gwyneplaine** [comment, marked as answer]
We have [a guide](https://github.com/keystonejs/keystone/blob/6ecb99a5f2d8a158b3d22036168b266db844a25c/docs/pages/docs/walkthroughs/using-keystones-api-from-a-web-app.mdx) in the works to make this clearer, at the moment its WIP, however it should have enough meat on the bone to help you get started.
